lib/dns-resolver.js

```diff
@@ -0,0 +1,501 @@
+'use strict'
+
+// source: https://raw.githubusercontent.com/szmarczak/cacheable-lookup/9e60c9f6e74a003692aec68f3ddad93afe613b8f/source/index.mjs
+
+/**
+
+MIT License
+
+Copyright (c) 2019 Szymon Marczak
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+
+*/
+
+const {
+  V4MAPPED,
+  ADDRCONFIG,
+  ALL,
+  promises: dnsPromises,
+  lookup: dnsLookup
+} = require('node:dns')
+const { promisify } = require('node:util')
+const os = require('node:os')
+
+const { Resolver: AsyncResolver } = dnsPromises
+
+const kExpires = Symbol('expires')
+const roundRobinStrategies = ['first', 'random']
+
+const supportsALL = typeof ALL === 'number'
+
+const map4to6 = (entries) => {
+  for (const entry of entries) {
+    if (entry.family === 6) {
+      continue
+    }
+
+    entry.address = `::ffff:${entry.address}`
+    entry.family = 6
+  }
+}
+
+const getIfaceInfo = () => {
+  let has4 = false
+  let has6 = false
+
+  for (const device of Object.values(os.networkInterfaces())) {
+    for (const iface of device) {
+      if (iface.internal) {
+        continue
+      }
+
+      if (iface.family === 'IPv6') {
+        has6 = true
+      } else {
+        has4 = true
+      }
+
+      if (has4 && has6) {
+        return { has4, has6 }
+      }
+    }
+  }
+
+  return { has4, has6 }
+}
+
+const isIterable = (map) => {
+  return Symbol.iterator in map
+}
+
+const ignoreNoResultErrors = (dnsPromise) => {
+  return dnsPromise.catch((error) => {
+    if (
+      error.code === 'ENODATA' ||
+      error.code === 'ENOTFOUND' ||
+      error.code === 'ENOENT' // Windows: name exists, but not this record type
+    ) {
+      return []
+    }
+
+    throw error
+  })
+}
+
+const ttl = { ttl: true }
+const all = { all: true }
+const all4 = { all: true, family: 4 }
+const all6 = { all: true, family: 6 }
+
+class DNSResolver {
```

Comment:
I would prefer using private fields and methods here rather than using underscore-prefixed methods/props.

Reply:
Done